### PR TITLE
Add shuffle_table to table utilities

### PR DESCRIPTION
**Pull request by mheguy**

### CODE REVIEW

File: utils/table.lua [hunk 1]
```diff
@@ -102,6 +106,20 @@ table.get_random_weighted = function(weighted_table, item_index, weight_index)
     end
 end
 
+--- Creates a fisher-yates shuffle of a sequential number-indexed table
+-- from: http://www.sdknews.com/cross-platform/corona/tutorial-how-to-shuffle-table-items
+-- @param table takes a table
+table.shuffle_table = function(t)
+    local rand = math.random
```
Comment (linaori) [inline utils/table.lua:113]: This one can be replaced with `random`, which is already imported

File: utils/table.lua [hunk 2]
```diff
@@ -102,6 +106,20 @@ table.get_random_weighted = function(weighted_table, item_index, weight_index)
...
+    local iterations = #t
```
Comment (linaori) [inline utils/table.lua:114]: Can you `error()` if size is 0? This in case someone uses it with a non-sequentially indexed array

**Review by linaori — COMMENTED** [review]
Nice! I can probably put this to use somewhere in the future

**Comment by Valansch** [comment]
I suggest this change:

```lua
table.shuffle_table = function(t, rng)
    local rand = rng or math.random
```

File: utils/table.lua (hunk 1 above)
Comment (mheguy) [reply utils/table.lua:113]: Yeah that was silly of me. =)

File: utils/table.lua (hunk 2 above)
Comment (mheguy) [reply utils/table.lua:114]: Can do.

**Comment by mheguy** [comment]
> I suggest this change

Done

**Comment by mheguy** [comment]
Closing this PR as it will be included as part of another. The suggested changes have all been made.

**Pull request closed without merging**